### ISSUE TITLE
Add method AbstractBlock.append(AbstractNode) to add a child node to a block

### ISSUE THREAD

**Issue by robertpanzer** [asciidoctor/asciidoctorj#325]
See http://discuss.asciidoctor.org/AsciidoctorJ-Block-Processor-Problems-tp3150p3297.html:

As you can see from my previous message, we are missing two key methods in the AsciidoctorJ API.

First, we need to add the append method to the AbstractBlock interface. This method allows child blocks to be added to a block.

**Comment by robertpanzer** [comment]
@mojavelinux: Shouldn't this already work like this?

```
AbstractBlock newBlock = createBlock(block,...);
block.getBlocks().add(newBlock);
```

**Comment by mojavelinux** [comment]
Aha! That is true. I didn't think about that since I always use the append method internally in Asciidoctor core.

I think we should still add the method though. But it's good to know that it's no longer a showstopper.

**Comment by mojavelinux** [comment]
Keep in mind the append method is important because, in the future, we may augment the method to automatically reparent the node (just like in DOM).

**Comment by mojavelinux** [comment]
...and that would be way I am consistent about using the append method in core.

**Comment by robertpanzer** [comment]
Fixed with #337
